stan/math/prim/fun/gaus_interp.hpp

```diff
@@ -0,0 +1,198 @@
+  // get lower and upper indexes for integration bounds
+  auto lb = upper_bound(xs.begin(), xs.end(), x - NSTDS * sig);
+  int ind_start = distance(xs.begin(), lb) - 1;
+  ind_start = std::max(0, ind_start);
+
+  auto ub = upper_bound(xs.begin(), xs.end(), x + NSTDS * sig);
+  int ind_end = distance(xs.begin(), ub);
+  ind_end = std::min(n - 1, ind_end);
+
+  // sum convolutions over intervals
+  using return_t = return_type_t<Tx>;
+  return_t y = 0;
+  for (int i = ind_start; i < ind_end; i++) {
+    y += conv_gaus_line(xs2[i], xs2[i + 1], params.as[i], params.bs[i], x,
+                        params.sig2);
+  }
+
+  return y;
+}
+
+/**
+ * This function was written to be used with gaus_interp. This function
+ * computes the shifted y-values of the reference points of an interpolation
+ * in such a way that when that piecewise linear function is convolved
+ * with a Gaussian kernel, the resulting function coincides with the
+ * points \f$(xs_i, ys_i)\f$ inputted into this function. The output of this
+ * function depends heavily on the choice of width of the Gaussian
+ * kernel, which at the time of writing, is set to one tenth the
+ * minimum distance between successive elements of the vector xs.
+ * A tolerance for the maximum distance between the interpolation and
+ * all reference points is also set manually and is not an input.
+ *
+ *
+ * @param xs vector of independent variable of reference points
+ * @param ys vector of dependent variable of reference points
+ * @return struct containing slopes, intercepts, and width of kernel
+ */
+gaus_interp_params gaus_interp_precomp(std::vector<double> const& xs,
+                                       std::vector<double> const& ys) {
+  static char const* function = "gaus_interp_precomp";
+  check_not_nan(function, "xs", xs);
+  check_not_nan(function, "ys", ys);
+  check_ordered(function, "xs", xs);
+
+  using internal::min_diff;
+  gaus_interp_params params;
+  const double INTERP_TOL = 1e-8;
+  const double SIG2_SCALE = 0.1;
+  int n = xs.size();
+
+  // find minimum distance between points for std of gaussian kernel
+  params.sig2 = stan::math::square(min_diff(n, xs) * SIG2_SCALE);
+  params.as.resize(n - 1);
+  params.bs.resize(n - 1);
+
+  // copy ys into a new vector that will be changed
+  std::vector<double> y2s = ys;
+
+  // interatively find interpolation that coincides with ys at xs
+  int max_iters = 50;
+  double dmax, dd;
+  for (int j = 0; j < max_iters; j++) {
+    // find slope and intercept of line between each point
+    for (int i = 0; i < n - 1; i++) {
+      params.as[i] = (y2s[i + 1] - y2s[i]) / (xs[i + 1] - xs[i]);
+      params.bs[i] = -xs[i] * params.as[i] + y2s[i];
+    }
+
+    dmax = 0;
+    for (int i = 0; i < n; i++) {
+      dd = ys[i] - gaus_interp(xs, y2s, params, xs[i]);
+      y2s[i] += dd;
+      dmax = std::max(std::abs(dd), dmax);
+    }
+    if (dmax < INTERP_TOL)
+      break;
+  }
+
+  return params;
```

Comment:
[optional] personally I prefer constructing this at the end like

```suggestion
  return gaus_interp_params(as, bs, sig2);
```